### ISSUE TITLE
Muting users dynamically

### ISSUE THREAD

**Issue by naps62** [goinstant/webrtc#15]
I want to automatically control which users are muted/unmuted. In my particular use case, there is a user with more privileges, which should be allowed to manually select who is speaking at each time.

He should be allowed to "give voice" to a specific user, which would automatically trigger a mute of every other user, and an unmute of the chosen one

for that I need access to the `goRTC#mute` function
It seems I can get access to that object by calling `webrtc._controller._goRTC`, but given that underscore'd variables usually mean they are private, this raises some concerns

Should I really do this? Or is there a better way to achieve what I want?
If not, should a public method be added to get access to that object? It makes sense to me, and I could easily make the Pull Request for that, if there's no better solution

**Comment by colinmacdonald** [comment]
Makes sense to me. It's going to be a bit more involved than just muting all others because we need to establish permissions. Specifically, the widget would need a feature to obey a user with authority. This authority would probably take the form of a key on the user object paired with an [ACL](https://developers.goinstant.com/v1/security_and_auth/guides/creating_and_managing_acl.html) that prevents changing it (so other user's can't claim authority for themselves).

As for who gets this authority key, I see it working two ways:
1. The authority key would be set externally to the widget but have the widget optionally obey the authority key and manage the muted state of each user accordingly.
2. Provide a UI in the widget itself for establishing a new conference session given the user's ID who should claim authority.

I think I like the first one better because it fits more closely with how the widget was originally designed (i.e. pass it a room vs having the widget create and join rooms).

**Comment by naps62** [comment]
Oh, I guess I didn't explain myself very well. I wasn't actually thinking of doing that feature directly on the widget. I'm already implementing it on my own app, by using GoInstant Keys to define who is the currently speaking user (which should get sound, and be expanded on the view of every client)
I'm also not caring all that much about security right now, as this is embedded in a Rails application MVP, which already defines the role of each user, and sets up the view accordingly. Better security will be implemented after the initial MVP

My concern here was only with accessing the goRTC object, which seems to be hidden in "private" variables. A simple way of retrieving this object with a public method/variable seems to make sense.

**Comment by colinmacdonald** [comment]
Oh, i see what you need now. Yeah, the goRTC object is just an instance of [GoRTC](https://github.com/goinstant/gortc) you "should" be able to access that private object and use it, but I cant guarantee that the widget would function properly (specifically the state of the view).

You should also check out the [GoRTC](https://github.com/goinstant/gortc) library as you can create a more customized webrtc app from the ground up with it. The WebRTC widget is actually built on top of GoRTC.
